src/material-experimental/select/testing/select-harness.ts

```diff
@@ -85,12 +86,12 @@ export class MatSelectHarness extends ComponentHarness {
   }
 
   /** Gets the options inside the select panel. */
-  async getOptions(): Promise<TestElement[]> {
+  async getOptions(): Promise<MatSelectOptionHarness[]> {
```

Comment:
Did we want to make this optionally take a `HarnessPredicate`? I seem to remember discussing that at some point. So we can search for a specific option more easily

Reply:
I'm not sure what that would look like. Here's what I got to compile, but it feels a little weird so I don't know if it's correct:

```ts
async getOptions(predicate?: HarnessPredicate<MatSelectOptionHarness>): Promise<MatSelectOptionHarness[]> {
  return this._documentRootLocator.locatorForAll(predicate || MatSelectOptionHarness)();
}
```

Reply:
Sorry, I think I meant `OptionHarnessFilters` not `HarnessPredicate`. That way people can do `getOptions({text: 'Option 1'})`. Under the hood that would use a harness predicate: `...locatorForAll(MatSelectOptions.with(options))`. My bad for not explaining that right